Here is my individual work for Project Stage 1. For my individual work, I focused on ACS Social, Economic, and Housing. Within my Dataset, I will also be relating it to the Covid-19 Data for the state that I chose, which was Florida. 

My data will be able to help to see a contrast between the Social, Housing, and Economic status during this time of Covid-19. I predict that it will stay consistent/stable.

In [103]:
import pandas as pd 
import numpy as np 

# Create a dataframe from CSV file
file_path = r"ACSDP5Y2020.DP02-Data.csv"
data_df = pd.read_csv(file_path)

cases_file = r"covid_confirmed_usafacts.csv"
cases_df = pd.read_csv(cases_file) 
 
data_df.columns = data_df.iloc[0]
data_df = data_df.drop(0)
df = df.reset_index(drop=True) 

#Geography name will be changed in order to merge to CountyFIPS 
new_column_name = 'countyFIPS' 
data_df = data_df.rename(columns={'Geography': new_column_name}) 
data_df['countyFIPS'] = data_df['countyFIPS'].str[9:]  
data_df 


C:\Users\jenny\AppData\Local\Temp\ipykernel_5836\1177074973.py:6: DtypeWarning: Columns (2,3,4,5,8,9,12,13,16,17,18,19,20,21,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264

0    countyFIPS              Geographic Area Name  \
1         01001           Autauga County, Alabama   
2         01003           Baldwin County, Alabama   
3         01005           Barbour County, Alabama   
4         01007              Bibb County, Alabama   
5         01009            Blount County, Alabama   
...         ...                               ...   
3217      72145  Vega Baja Municipio, Puerto Rico   
3218      72147    Vieques Municipio, Puerto Rico   
3219      72149   Villalba Municipio, Puerto Rico   
3220      72151    Yabucoa Municipio, Puerto Rico   
3221      72153      Yauco Municipio, Puerto Rico   

0    Estimate!!HOUSEHOLDS BY TYPE!!Total households  \
1                                             21559   
2                                             84047   
3                                              9322   
4                                              7259   
5                                             21205   
...                                             ...   
3217                                            NaN   
3218                                            NaN   
3219                                            NaN   
3220                                            NaN   
3221                                            NaN   

0    Margin of Error!!HOUSEHOLDS BY TYPE!!Total households  \
1                                                   366      
2                                                  1143      
3                                                   338      
4                                                   299      
5                                                   430      
...                                                 ...      
3217                                                NaN      
3218                                                NaN      
3219                                                NaN      
3220                                                NaN      
3221                                                NaN      

0    Estimate!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household  \
1                                                 11281                         
2                                                 46277                         
3                                                  3613                         
4                                                  3597                         
5                                                 12053                         
...                                                 ...                         
3217                                                NaN                         
3218                                                NaN                         
3219                                                NaN                         
3220                                                NaN                         
3221                                                NaN                         

0    Margin of Error!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household  \
1                                                   552                                
2                                                  1550                                
3                                                   305                                
4                                                   382                                
5                                                   675                                
...                                                 ...                                
3217                                                NaN                                
3218                                                NaN                                
3219                                                NaN                                
3220                                                NaN                                
3221            

In [113]:
import pandas as pd
cases_data = pd.read_csv('covid_confirmed_usafacts.csv') 
cases_df = pd.DataFrame(cases_data) 

cases_df[cases_df.loc[:, 'State'] == 'FL']


countyFIPS            County Name State  StateFIPS  2020-01-22  \
329           0  Statewide Unallocated    FL         12           0   
330       12001        Alachua County     FL         12           0   
331       12003          Baker County     FL         12           0   
332       12005            Bay County     FL         12           0   
333       12007       Bradford County     FL         12           0   
..          ...                    ...   ...        ...         ...   
392       12125          Union County     FL         12           0   
393       12127        Volusia County     FL         12           0   
394       12129        Wakulla County     FL         12           0   
395       12131         Walton County     FL         12           0   
396       12133     Washington County     FL         12           0   

     2020-01-23  2020-01-24  2020-01-25  2020-01-26  2020-01-27  ...  \
329           0           0           0           0           0  ...   
330           0           0           0           0           0  ...   
331           0           0           0           0           0  ...   
332           0           0           0           0           0  ...   
333           0           0           0           0           0  ...   
..          ...         ...         ...         ...         ...  ...   
392           0           0           0           0           0  ...   
393           0           0           0           0           0  ...   
394           0           0           0           0           0  ...   
395           0           0           0           0           0  ...   
396           0           0           0           0           0  ...   

     2023-07-14  2023-07-15  2023-07-16  2023-07-17  2023-07-18  2023-07-19  \
329       20363       20363       20363       20363       20363       20363   
330       90634       90634       90634       90634       90634       90634   
331       10787       10787       10787       10787       10787       10787   
332       55836       55836       55836       55836       55836       55836   
333        9324        9324        9324        9324        9324        9324   
..          ...         ...         ...         ...         ...         ...   
392        5140        5140        5140        5140        5140        5140   
393      152326      152326      152326      152326      152326      152326   
394       12314       12314       12314       12314       12314       12314   
395       19888       19888       19888       19888       19888       19888   
396        7731        7731        7731        7731        7731        7731   

     2023-07-20  2023-07-21  2023-07-22  2023-07-23  
329       20363       20363       20363       20363  
330       90634       90634       90634       90634  
331       10787       10787       10787       10787  
332       55836       55836       55836       55836  
333        9324        9324        9324        9324  
..          ...         ...         ...         ...  
392        5140        5140        5140        5140  
393      152326      152326      152326      152326  
394       12314       12314       12314       12314  
395       19888       19888       19888       19888  
396        7731        7731        7731        7731  

[68 rows x 1269 columns]

As we can see, I chose to analyze the state of Florida. We can tell that it is stable since the "statewide unallocated" has the number 20363 repeated multiple times. As predicted, yes it was stable.

In [127]:
import pandas as pd 

data_df['countyFIPS'] = data_df['countyFIPS'].astype(int) 

covid_data = pd.read_csv('final_merged_data.csv') 
covid_data_df = pd.DataFrame(covid_data) 

merged_data = pd.merge(data_df, cases_df, on='countyFIPS', how='outer')
merged_data.head()

countyFIPS Geographic Area Name  \
0           0                  NaN   
1           0                  NaN   
2           0                  NaN   
3           0                  NaN   
4           0                  NaN   

  Estimate!!HOUSEHOLDS BY TYPE!!Total households  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4                                            NaN   

  Margin of Error!!HOUSEHOLDS BY TYPE!!Total households  \
0                                                NaN      
1                                                NaN      
2                                                NaN      
3                                                NaN      
4                                                NaN      

  Estimate!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household  \
0                                                NaN                         
1                                                NaN                         
2                                                NaN                         
3                                                NaN                         
4                                                NaN                         

  Margin of Error!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household  \
0                                                NaN                                
1                                                NaN                                
2                                                NaN                                
3                                                NaN                                
4                                                NaN                                

  Estimate!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household!!With children of the householder under 18 years  \
0                                                NaN                                                                          
1                                                NaN                                                                          
2                                                NaN                                                                          
3                                                NaN                                                                          
4                                                NaN                                                                          

  Margin of Error!!HOUSEHOLDS BY TYPE!!Total households!!Married-couple household!!With children of the householder under 18 years  \
0                                                NaN                                                                                 
1                                                NaN                                                                                 
2                                                NaN                                                                                 
3                                                NaN                                                                                 
4                                                NaN                                                                                 

  Estimate!!HOUSEHOLDS BY TYPE!!Total households!!Cohabiting couple household  \
0                                                NaN                            
1                                                NaN                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

  Margin of Error!!HOUSEHOLDS BY TYPE!!Total households!!Cohabiting cou